apis/deployer/healthchecks/validation/validation.go

```diff
@@ -0,0 +1,121 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package validation
+
+import (
+	"errors"
+
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/client-go/util/jsonpath"
+
+	"github.com/gardener/landscaper/apis/deployer/healthchecks"
+)
+
+// ValidateHealthCheckConfiguration validates a healthcheck configuration
+func ValidateHealthCheckConfiguration(fldPath *field.Path, config *healthchecks.HealthChecksConfiguration) field.ErrorList {
+	var allErrs field.ErrorList
+
+	// if we have a custom healthcheck configuration, the default should be disabled
+	disableDefault := config.DisableDefault
+	customHealthChecks := config.CustomHealthChecks
+	if !disableDefault && customHealthChecks != nil && len(customHealthChecks) > 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("customHealthChecks"), disableDefault, "custom health checks require disabling the default health check"))
```

Comment:
why do we require the default to be disabled?

I think its valid to say that I still want default healthchecks for deployment and still have custom healthchecks for crds.
Maybe we should also make the default healthchecks configurable per resource? WDYT?

Reply:
Yes, totally makes sense. I remove this piece of code from the validation and changed the `ensure.go` of both, Manifest and Helm deployer so that default and custom healthchecks can run together.